critiquebrainz/db/review.py

```diff
@@ -613,22 +626,32 @@ def delete(review_id):
         db_avg_rating.update(review["entity_id"], review["entity_type"])
 
 
-def distinct_entities():
+def distinct_entities(*, trans_conn=None):
```

Comment:
I'm not entirely happy with the `if-else` conditions here. Can we do something like this? Do the meat of the stuff in a new function which takes a connection as a parameter, change the original functions to use the new one and use the new one directly in the dump code?

```
def get_distinct_entities(connection):
     query = sqlalchemy.text("""
        SELECT DISTINCT entity_id
          FROM review
    """)
    results = connection.execute(query)
    return {row[0] for row in results.fetchall()}
   
def distinct_entities():
    with db.engine.connect() as connection:
        return get_distinct_entities(connection)
```

Reply:
This would also mean that we don't _have_ to worry about breaking existing code.

Reply:
yes, so we'd just use `get_distinct_entities(transaction)` in the `json()` function in `dump_manager.py`

Reply:
@paramsingh it would work for functions where only `connection` object has to be passed. What about functions where multiple arguments have to be passed and all these arguments are used to formulate the query?

For example, have a look at [list_reviews()](https://github.com/metabrainz/critiquebrainz/blob/master/critiquebrainz/db/review.py#L319). The `list_review()` function is being called by `json()` function by supplying multiple arguments which are then used to formulate a query. Therefore, if we make a new `get_list_reviews(connection)` function, we cannot formulate query inside it without all the arguments of `list_reviews()` function.

Reply:
Why not make a get_reviews(connection, *, inc_drafts...) and then call it directly in the json function and the list_reviews function